hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -68,6 +71,10 @@ public StructuredTrace getTrace() {
     return apiNodeEventEdgeList;
   }
 
+  public Integer getApiNodeIndexForEvent(ByteBuffer eventId) {
```

Comment:
Why get an index rather than the node itself? Why do we even work in indexes in this file at all when we have maps?

Reply:
The reason is that ApiNodeEventEdge which represents an api edge, uses ApiNode index as src & tgt instead of Object itself. This class thus maintains `Map<ApiNode<Event>, Integer> apiNodeToIndex` & `List<ApiNode<Event>> nodeList`. 
Also, with the way ApiNode is defined, I feel it shouldn't be used as map key, ideally map keys should be immutable (preferably cache its hash).
Getting rid of these index based usage, can be taken up as part of https://github.com/hypertrace/hypertrace-ingester/issues/139. 
Please add your suggestion on that issue.

Reply:
So an ApiNode is mutable? Why is that? If changing the new APIs/code requires extensive work in existing code, I'm fine with deferring it to a larger refactor as part of #139 , but if it can be changed on its own to make that part of the API more sensible, I don't see the benefit in deferring it.

Reply:
I realised that this new api was only used in Test, so just removed it. 

> So an ApiNode is mutable? Why is that? 
I am not sure, but in the code we are not mutating it post instantiation.